src/components/playlist/panel/playlist-panel.js

```diff
@@ -0,0 +1,83 @@
+import videojs from 'video.js';
+import 'assets/styles/components/playlist.scss';
+import PlaylistPanelItem from './playlist-panel-item';
+
+const dom = videojs.dom || videojs;
+
+const Component = videojs.getComponent('Component');
+
+class PlaylistPanel extends Component {
+
+  constructor(player, options = {}) {
+    super(player, options);
+
+    const itemChangeHandler = () => {
+      this.render();
+    }
+
+    player.on('playlistitemchanged', itemChangeHandler);
+
+    this.render();
+
+    this.dispose = () => {
+      super.dispose();
+      player.off('playlistitemchanged', itemChangeHandler);
+    } 
+  }
+
+  createEl() {
+    const el = super.createEl();
+    el.classList.add('cld-plw-panel');
+
+    return el;
+  }
+
+  removeAll() {
+    const childrens = this.children();
+    for (let i = childrens.length - 1; i >= 0; --i) {
+      this.removeChild(childrens[i]);
+    }
+  }
+
+  getItems() {
+    const playlist = this.player().cloudinary.playlist();
+
+    if (this.options_.showAll) {
+      return playlist.list();
+    }
+
+    const startIndex = playlist.currentIndex() + 1;
+    const endIndex = startIndex + this.options_.total;
+    let selectedItems = playlist.list().slice(startIndex, endIndex);
+
+    if (selectedItems.length < this.options_.total) {
+      selectedItems = [...selectedItems, ...playlist.list().slice(0, this.options_.total - selectedItems.length)];
```

Comment:
Currently, for the case of total items of 6 and only 2 items in the playlist, the widget will show: 
item1, item2, item1, item2

which is incorrect. Consider using the following logic:


```javascript
        const items = [];

        const numOfItems = this.options_.total;

        while (items.length < numOfItems) {
          index = playlist.nextIndex(index);
          if (index === -1) {
            index = 0;
          }

          const source = playlist.list()[index];
          items.push(source);
        }

        return items;
      };
```